infrastructure/src/main/java/org/corfudb/infrastructure/ServerContext.java

```diff
@@ -428,6 +432,25 @@ public Layout getManagementLayout() {
         return dataStore.get(Layout.class, PREFIX_MANAGEMENT, MANAGEMENT_LAYOUT);
     }
 
+    /**
+     * Sets the log unit reset epoch.
+     *
+     * @param resetEpoch Epoch at which the reset command was received.
+     */
+    public synchronized void setLogUnitResetEpoch(long resetEpoch) {
+        dataStore.put(Long.class, PREFIX_LOGUNIT, RESET_EPOCH, resetEpoch);
+    }
+
+    /**
+     * Fetches the epoch at which the last reset command was received.
+     *
+     * @return Reset epoch.
+     */
+    public synchronized long getLogUnitResetEpoch() {
+        Long resetEpoch = dataStore.get(Long.class, PREFIX_LOGUNIT, RESET_EPOCH);
+        return resetEpoch == null ? Layout.INVALID_EPOCH : resetEpoch;
```

Comment:
how does invalid epoch compare with normal epochs.

Reply:
INVALID epoch will always be less than any layout epoch.
INVALID_EPOCH = -1